static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -0,0 +1,448 @@
+    citeImage: ?String
+) =>
+    template(acquisitionsTestimonialBlockTemplate, {
+        quoteSvg: testimonialParameters.quoteSvg,
+        testimonialMessage: testimonialParameters.testimonialMessage,
+        testimonialName: testimonialParameters.testimonialName,
+        citeImage,
+    });
+
+const defaultPageCheck = (page: Object): boolean =>
+    page.contentType === 'Article' && !page.isMinuteArticle;
+
+const defaultCanEpicBeDisplayed = (
+    campaignId: string,
+    showToContributorsAndSupporters: boolean,
+    useTargetingTool: boolean,
+    // set this default here as well as makeABTest so that this function is more usable externally
+    pageCheck: (page: Object) => boolean = defaultPageCheck,
+    locations: Array<string> = [],
+    locationCheck: (location: string) => boolean = () => true
+): boolean => {
+    const canReasonablyAskForMoney =
+        showToContributorsAndSupporters ||
+        commercialFeatures.canReasonablyAskForMoney;
+
+    const worksWellWithPageTemplate = pageCheck(config.page);
+
+    const storedGeolocation = geolocationGetSync();
+    const inCompatibleLocation = locations.length
+        ? locations.some(geo => geo === storedGeolocation)
+        : true;
+
+    const tagsMatch = doTagsMatch(campaignId, useTargetingTool);
+
+    return (
+        canReasonablyAskForMoney &&
+        worksWellWithPageTemplate &&
+        inCompatibleLocation &&
+        locationCheck(storedGeolocation) &&
+        tagsMatch
+    );
+};
+
+const getCopy = (useTailor: boolean): Promise<AcquisitionsEpicTemplateCopy> => {
+    if (useTailor) {
+        return isRegular().then(
+            regular =>
+                regular ? acquisitionsCopyRegulars : acquisitionsCopyControl
+        );
+    }
+
+    return new Promise(resolve => resolve(acquisitionsCopyControl));
+};
+
+const getCampaignCode = (
+    campaignCodePrefix,
+    campaignID,
+    id,
+    campaignCodeSuffix
+) => {
+    const suffix = campaignCodeSuffix ? `_${campaignCodeSuffix}` : '';
+    return `${campaignCodePrefix}_${campaignID}_${id}${suffix}`;
+};
+
+const addTrackingCodesToUrl = (base: string, campaignCode: string) => {
+    const params = {
+        REFPVID: (config.ophan && config.ophan.pageViewId) || 'not_found',
+        INTCMP: campaignCode,
+    };
+
+    return `${base}?${constructURLQuery(params)}`;
+};
+
+const makeEvent = (id: string, event: string): string => `${id}:${event}`;
+
+const registerIframeListener = (iframeId: string) => {
+    window.addEventListener('message', message => {
+        const iframe = document.getElementById(iframeId);
+
+        if (iframe) {
+            try {
+                const data = JSON.parse(message.data);
+
+                if (data.type === 'set-height' && data.value) {
+                    iframe.style.height = `${data.value}px`;
+                }
+            } catch (e) {
+                // Apparently I need a comment here to satisfy the linter
+                // TODO: change linter rule??
```

Comment:
I still think the `no-empty` rule has value, despite the awkwardness here. It forces you to be explicit about `catch` blocks that simply swallow exceptions, and ensures exceptions aren't swallowed by accident. It also ensures there are no empty `if` or loop blocks as a result of incomplete refactoring. 

There's a lot of discussion at eslint/eslint#1841, but I'm okay with the comment workaround. Happy to discuss if I've missed a potential issue